crates/ra_hir_ty/src/lib.rs

```diff
@@ -877,6 +877,58 @@ impl Ty {
             _ => None,
         }
     }
+
+    pub fn impl_trait_ref(&self, db: &dyn HirDatabase) -> Option<TraitRef> {
```

Comment:
impl Trait can have multiple trait ref bounds; ideally I think this should just be a function that returns the bounds